packages/flutter/lib/src/services/lifecycle.dart

```diff
@@ -0,0 +1,32 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+
+typedef void ShutdownListener();
+
+class _Lifecycle {
+  _Lifecycle() {
+    ui.lifecycleHooks.onShutdown = _handleShutdown;
+  }
+
+  final List<ShutdownListener> _shutdownListeners = new List<ShutdownListener>();
```

Comment:
Maybe this should be a LinkedHashSet?  What you've got here is n^2 in the number of in-flight mojo requests.